src/ion/moves.rs

```diff
@@ -436,18 +436,16 @@ impl<'a, F: Function> Env<'a, F> {
                                 );
                                 #[cfg(debug_assertions)]
                                 {
-                                    if log::log_enabled!(log::Level::Trace) {
-                                        self.annotate(
-                                            self.cfginfo.block_entry[block.index()],
-                                            format!(
-                                                "blockparam-in: block{} to block{}:into v{} in {}",
-                                                from_block.index(),
-                                                to_block.index(),
-                                                to_vreg.index(),
-                                                alloc
-                                            ),
-                                        );
-                                    }
+                                    self.annotate(
+                                        self.cfginfo.block_entry[block.index()],
+                                        format!(
```

Comment:
I think this change will mean that in `debug_assertions` builds with logging disabled, we now invoke this `format!()` call and allocate a string where we didn't before? Probably it should be guarded with the `trace_enabled!()` conditional as above...

Reply:
Ah you're right that it will allocate; to keep on using only the same single guard, i think it should be guarded with `annotations_enabled` too. Thanks!